replication/binlogsyncer.go

```diff
@@ -707,6 +707,11 @@ func (b *BinlogSyncer) parseEvent(s *BinlogStreamer, data []byte) error {
 	//skip OK byte, 0x00
 	data = data[1:]
 
+	var (
+		currentGTID string
```

Comment:
should this value be cached in `BinlogSyncer` to be used in the following binlog event?

Reply:
I think no need, it always retry from a gtid

Reply:
but `currentGTID` is in the function `parseEvents` scope, `b.gset` is not correctly updated then.

Reply:
I dont know what's meaning of you

Reply:
You can give a binlog event temporal logic example

Reply:
Does code `event.GSet, err = b.updateGtidSet(currentGTID)` in L743 and L745 want to update with the `currentGTID` that has the `gtid` value right in the same transaction?
Such as when processing the Query event for DDL `create database sharding1`, `b.gset` should be updated with `9afe121c-40c2-11e9-9ec7-0242ac110002:3`
```
| bin.000001 | 522 | Gtid           |       101 |         587 | SET @@SESSION.GTID_NEXT= '9afe121c-40c2-11e9-9ec7-0242ac110002:3'                                                                                                        |
| bin.000001 | 587 | Query          |       101 |         698 | create database `sharding1`
```

Reply:
Yep, then would it only receives a partial transaction?